pkg/user/handler.go

```diff
@@ -265,3 +269,52 @@ func (h Handler) FindAll(c *gin.Context) {
 
 	c.JSON(http.StatusOK, users)
 }
+
+// Delete user
+func (h Handler) Delete(c *gin.Context) {
+	// swagger:route DELETE /users/{id} deleteUser
+	//
+	// Delete user
+	//
+	// Delete user by id
+	//
+	// Security:
+	//	oauth2:
+	//
+	// Responses:
+	//	202:
+	//	401: Error
+	//	403: Error
+	//	404: Error
+	//	415: Error
+	id, ok := handler.GetPathParameter(c, "id")
+	if !ok {
+		return
+	}
+
+	user, err := handler.GetUserFromContext(c)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	_, err = h.userService.FindById(uint(id))
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	if user.ID == uint(id) {
+		forbiddenRequest := errdef.NewBadRequest("cannot delete the current user")
+		_ = c.Error(forbiddenRequest)
```

Comment:
if you wanted to return a https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403
you would need to do

`c.AbortWithError(http.StatusForbidden, errors.New("cannot delete the current user"))`

Reply:
I actually forgot to update the error variable name after changing the status code to Bad Request, as also suggested by @tonsV2 [here](https://github.com/dhis2-sre/im-manager/pull/257#issuecomment-1573155335). 😅